### PR TITLE
feat: Shape context

### PR DESCRIPTION
**Pull request by srednicki95**

### CODE REVIEW

File: src/components/AzureMapFeature/AzureMapFeature.tsx [hunk 1]
```diff
@@ -22,10 +25,35 @@ const AzureMapFeature = memo((props: IAzureMapFeature) => {
...
+    if (shapeRef && setCoords) {
+      shapeRef.setCoordinates(setCoords)
+    }
+  }, [setCoords])
+
+  useEffect(() => {
+    if (shapeRef && setProperties) {
+      shapeRef.setProperties(setProperties)
+    }
+  }, [setProperties])
```
Comment (msasinowski) [inline src/components/AzureMapFeature/AzureMapFeature.tsx]: We don't want to check this is somehow better ? like check if nested properties changed not only `ref`

File: src/types.ts [hunk 2]
```diff
@@ -182,6 +190,11 @@ export type IAzureLayerStatefulProviderProps = {
   events?: IAzureMapLayerEvent | any
   lifecycleEvents?: IAzureMapLifecycleEvent | any
 }
+export type IAzureShapeStatefulProviderProps = {
+  id?: string
+  properties?: any
```
Comment (msasinowski) [inline src/types.ts]: Did we don't have ony properties types for that if not use `Options`

File: src/types.ts [hunk 3]
```diff
@@ -124,9 +124,17 @@ export type IAzureMapLayerContextState = {
   layerRef: atlas.layer.SymbolLayer | atlas.layer.ImageLayer | atlas.layer.TileLayer | null
 }
 
+export type IAzureMapShapeContextState = {
+  shapeRef: atlas.Shape | null
+  createShape?: any
```
Comment (msasinowski) [inline src/types.ts]: this should be a typed function

File: src/contexts/AzureMapShapeContext.tsx [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+  id,
+  properties,
+  children
+}: IAzureShapeStatefulProviderProps) => {
+  const { mapRef } = useContext<IAzureMapsContextProps>(AzureMapsContext)
+  const { dataSourceRef } = useContext<IAzureMapDataSourceProps>(AzureMapDataSourceContext)
+  const [shapeRef, setShapeRef] = useState<atlas.Shape | any>(null)
+
+  const createShape = (
+    data: atlas.data.Feature<atlas.data.Geometry, any> | atlas.data.Geometry | any
```
Comment (msasinowski) [inline src/contexts/AzureMapShapeContext.tsx]: Instead of any maybe use Object

File: src/components/AzureMapFeature/AzureMapFeature.tsx [hunk 5]
```diff
@@ -22,10 +25,35 @@ const AzureMapFeature = memo((props: IAzureMapFeature) => {
     }
   }, [])
 
+  useEffect(() => {
```
Comment (msasinowski) [inline src/components/AzureMapFeature/AzureMapFeature.tsx]: My proposition here is to add optional type to azure map feature like type shape | feature. When type is a shape we create shape and make reactions if properties of coordinates are changed. So we accomplish that without implementing another context.

Comment (msasinowski) [reply src/components/AzureMapFeature/AzureMapFeature.tsx]: https://github.com/WiredSolutions/react-azure-maps/pull/35/files

Comment (srednicki95) [reply src/components/AzureMapFeature/AzureMapFeature.tsx]: Let's go that way! 🔥

**Comment by srednicki95** [comment]
Shape will implement in: https://github.com/WiredSolutions/react-azure-maps/pull/35

**Pull request closed without merging**